core-bundle/src/Resources/contao/models/ContentModel.php

```diff
@@ -475,6 +475,86 @@ public static function countPublishedByPidAndTable($intPid, $strParentTable, arr
 
 		return static::countBy($arrColumns, array($intPid, $strParentTable), $arrOptions);
 	}
+
+	/**
+	 * Find all visible content elements by their parent ID and parent table
+	 *
+	 * @param integer $intPid         The article ID
+	 * @param string  $strParentTable The parent table name
+	 * @param array   $arrOptions     An optional options array
+	 *
+	 * @return Collection|ContentModel[]|ContentModel|null A collection of models or null if there are no content elements
+	 */
+	public static function findVisibleByPidAndTable($intPid, $strParentTable, array $arrOptions=array())
```

Comment:
The naming could be misinterpreted I think. How about something like `findVisibleAndUpcomingByPidAndTable`?

Reply:
Or even `findPublishedAndUpcoming` and then we drop the original `findPublished`in 5? :see_no_evil:.